summary-visualiser/generate.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+set -euo pipefail
+
+SCRIPT_DIR=$(dirname "$0")
+# Read either from a filename or stdin.
+INPUT_JSON=$(cat "${1:-/dev/stdin}")
+# The scenario name is pulled from the JSON itself.
+# We'll need this later to fetch the right template.
+SCENARIO_NAME=$(echo "$INPUT_JSON" | jq -r ".run_summary.scenario_name")
```

Comment:
This might work with one element, but the content of this is an array isn't it? At least that's what I have in my latest generated data:

```json
[
  {
    "run_summary": {
      "run_id": "CKYEOs9IAH5W2n4s2TyXd",
      "scenario_name": "zero_arc_create_data",
      "started_at": 1762773592,
      "run_duration": 300,
      "peer_count": 3,
      "peer_end_count": 3,
      "assigned_behaviours": {
        "zero": 1,
        "full": 2
      },
```

Reply:
I think the problem here is trying to determine the scenario name in the script. That's not what I've been suggesting.

If instead there is a single top level template that contains something like this pseudo-code:

```
{foreach scenario in . }
  {if scenario.run_summary.scenario_name == "dht_sync_lag" }
    {renderDhtSyncLag scenario }
  {else}
    <p>No template for scenario {scenario.run_summary.scenario_name}</p>
  {/end}
{/end}
```

Then less work is needed in the script and the problem of handling multiple scenarios is set up ready for working with multiple and the script doesn't need adjusting later.

Reply:
https://github.com/holochain/wind-tunnel/pull/327/commits/2dd03f1b2e7f5a5b361eb2dabb0e24061866ce70

Note that this commit doesn't iterate through all the known scenario template names; instead, by convention, it loads up all the files in `templates/scenarios/` as helper templates, and hard-fails if one doesn't exist.

I hope this blend of convention and 'strong typing' isn't weird -- I figure the former reduces long-term maintenance and diff noise, while the latter is already true about the individual scenarios (it assumes a certain data structure and fails if it encounters something unexpected).